Description: zbar: tests: Handle symbol-bytes as not null-terminated
 Backport https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/4592 to
 avoid build failures on toolchains that expose this bug.
Origin: upstream, https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/5a6ceaf2f3ff7dc8f3cf34c1a001ed2a11a14fce
Bug: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/4592
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/tests/check/elements/zbar.c
+++ b/tests/check/elements/zbar.c
@@ -85,6 +85,8 @@
   int qual;
   const GValue *value;
   GBytes *symbol_bytes;
+  gconstpointer bytes_data;
+  gsize bytes_size;
 
   pipeline = setup_pipeline ();
 
@@ -112,8 +114,10 @@
   value = gst_structure_get_value (s, "symbol-bytes");
   fail_unless (G_VALUE_HOLDS (value, G_TYPE_BYTES));
   symbol_bytes = g_value_get_boxed (value);
-  fail_unless_equals_string (g_bytes_get_data (symbol_bytes, 0),
-      "9876543210128");
+  bytes_data = g_bytes_get_data (symbol_bytes, &bytes_size);
+  fail_unless_equals_uint64 (bytes_size, 13);
+  fail_unless (memcmp (bytes_data, "9876543210128", 13) == 0,
+      "Bad symbol-bytes data: %.13s", bytes_data);
 
   fail_if (gst_structure_has_field (s, "frame"));
 
